### ISSUE TITLE
No handler could be found for the page 'index.php?q=regen_thumb/one/28583'

### ISSUE THREAD

**Issue by diamondsw** [shish/shimmie2#1029]
**Server Software**
about:
  title: "HomeBooru"
  theme: "danbooru2"
  url: "http://[redacted]/index.php?q="

versions:
  shimmie: "v2.10.3-20240211-62ece58"
  schema: 21
  php: "8.2.7"
  db: "pgsql PostgreSQL 11.22 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20231014) 13.2.1 20231014, 64-bit"
  os: "Linux e2b4ff35dcec 6.1.0-17-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30) x86_64"
  server: "Unit/1.31.1"

extensions:
  core: ["admin","alias_editor","bbcode","comment","download","et","ext_manager","four_oh_four","handle_pixel","help_pages","image","index","media","mime","replace_file","setup","static_files","system","tag_edit","tag_list","upgrade","upload","user","user_config","view"]
  extra: ["auto_tagger","autocomplete","bulk_add_csv","emoticons_list","et_server","favorites","home","pools","post_titles","random_image","random_list","rating","regen_thumb","relationships","tag_categories","tag_tools","tagger_xml"]
  handled_mimes: ["image/jpeg","image/gif","image/png","image/webp"]

stats:
  images: 28132
  comments: 0
  users: 2

media:
  memory_limit: "32MB"
  disk_use: "129GB"
  disk_total: "197GB"

thumbnails:
  engine: "convert"
  quality: 75
  width: 190
  height: 300
  scaling: 100
  mime: "image/jpeg"

**Client Software (please complete the following information)**
 - Device: macOS Desktop
 - Browser: Safari

**What steps trigger this bug**
1. Go to any post
2. Under "Post Controls", click "Regenerate Thumbnail"

**What did you expect to happen?**
Thumbnail should be regenerated.

**What actually happened?**
`No handler could be found for the page 'index.php?q=regen_thumb/one/28583'`

Feels related to #1026

**Comment by shish** [comment]
> Feels related to https://github.com/shish/shimmie2/issues/1026

Yup - trying to fix thing a broke thing b, fixing thing b broke thing c, at this point I am pretty much rewriting the whole URL generation-and-parsing code but this time being aware of all the edge cases and writing unit tests first ^^;;

(Much of this code has been largely unchanged for like 15 years...)
